### PR TITLE
added, changed and corrected stuff

### PR DESCRIPTION
**Pull request by NotAnUser1**
added option to show teammate buildings without enabling teammate (player) esp
added objectives esp
added disguised cond esp
added more observer modes to spectator list and made it team colored
added flask style to magic spells esp
added "hide invisible" to prevent drawing on invis players
added edibles/lunchboxes support to healthpack esp
added more checks to auto strafer to prevent it from working under certain conditions
corrected(?) vaccinator's resistance conds
corrected a typo in auto pyro
changed explosive esp code to make it look neater
changed healthpack and ammopack code to make use of switch statements
changed vaccinator charges code to get rid of unneeded characters

### CODE REVIEW

**Comment by BenCat07** [comment]
Tell me once this is actually ready, right now you seem to still be adding stuff

**Comment by TotallyNotElite** [comment]
this PR is hella broken, you need to rebase it

File: src/hacks/ESP.cpp [hunk 1]
```diff
@@ -1506,7 +1598,7 @@ void _FASTCALL ProcessEntity(CachedEntity *ent)
                                         AddEntityString(ent, charge + "% Uber", colors::Health(floor(CE_FLOAT(weapon, netvar.m_flChargeLevel) * 100), 100));
                                     }
                                     else
-                                        AddEntityString(ent, charge + "% Uber | Charges: " + std::to_string(floor(CE_FLOAT(weapon, netvar.m_flChargeLevel) / 0.25f)), colors::Health((CE_FLOAT(weapon, netvar.m_flChargeLevel) * 100), 100));
+                                        AddEntityString(ent, charge + "% Uber | Charges: " + std::to_string((int) floor((CE_FLOAT(weapon, netvar.m_flChargeLevel) / 0.25f) * 100) / 100), colors::Health((CE_FLOAT(weapon, netvar.m_flChargeLevel) * 100), 100));
```
Comment (TotallyNotElite) [inline src/hacks/ESP.cpp:1601]:
```suggestion
                                        AddEntityString(ent, charge + "% Uber | Charges: " + std::to_string((int) floor(CE_FLOAT(weapon, netvar.m_flChargeLevel) / 0.25f), colors::Health((CE_FLOAT(weapon, netvar.m_flChargeLevel) * 100), 100));
```

This makes more sense IMO

File: src/hacks/Misc.cpp [hunk 2]
```diff
@@ -369,22 +371,30 @@ void Draw()
...
                     break;
                 case 7:
-                    observermode = "Freecam";
+                    observermode = "FreeCam";
                     break;
                 }
-                AddSideString(format(info.name, " ", observermode));
+                rgba_t color = ent->m_iTeam() == TEAM_BLU ? colors::blu : (ent->m_iTeam() == TEAM_RED ? colors::red : colors::white);
+
+                AddSideString(format(info.name, " - (", observermode, ")"), color);
```
Comment (TotallyNotElite) [inline src/hacks/Misc.cpp:397]: IMO this should not be changed. Current version is fine. Deathcam, freezcam, etc are not very useful

File: src/hacks/Radar.cpp [hunk 3]
```diff
@@ -107,6 +108,8 @@ void DrawEntity(int x, int y, CachedEntity *ent)
         {
             if (CE_BYTE(ent, netvar.iLifeState))
                 return; // DEAD. not big surprise.
```
Comment (TotallyNotElite) [inline src/hacks/Radar.cpp:110]:
```suggestion
            if (ent->m_bAlivePlayer())
                return; // DEAD. not big surprise.
```

Not related to your code, but I still need to fix this

**Review by TotallyNotElite — CHANGES_REQUESTED** [review]

File: data/menu/nullifiedcat/visuals/esp.xml [hunk 4]
```diff
@@ -22,6 +23,7 @@
             <AutoVariable width="fill" target="esp.item.money-red" label="MvM red money" tooltip="Red money is automatically collected."/>
             <AutoVariable width="fill" target="esp.item.powerup" label="Mannpower powerups"/>
             <AutoVariable width="fill" target="esp.item.spellbook" label="Spellbooks"/>
+            <AutoVariable width="fill" target="esp.item.objectives" label="Objectives" tooltip="Intels, carts and mvm bomb"/>
```
Comment (TotallyNotElite) [inline data/menu/nullifiedcat/visuals/esp.xml:26]:
```suggestion
            <AutoVariable width="fill" target="esp.item.objectives" label="Objectives" tooltip="Intel, carts and mvm bombs"/>
```

Learn to grammar

**Review by TotallyNotElite — DISMISSED** [review]

**Review by TotallyNotElite — APPROVED** [review]

**Review by BenCat07 — APPROVED** [review]
LGTM

**Pull request merged**